### DISCUSSION TITLE
Upload images as blobs

### DISCUSSION THREAD

**Discussion by MatthijsMali** [keystonejs/keystone#7531] [Questions] [answered]
I’m wondering how people are currently handling file and image uploads. 

In my solution there’s a separate docker container for my nextjs frontend and one for my keystone backend. Currently I’m thinking to have a mounted folder in my docker-compose that is referenced as a file upload location in my Keystone config. But I was hoping for something blob storage like, so that my images are in my postgress database. Has anyone got something like that working?

**Comment by MatthijsMali** [comment]
A little more background. I'm creating a website for my baseball team. It's a pretty basic Blog, Teams, Event Calendar, Scores setup. There are image uploads involved.

I currently have the following setup in my Docker environment:

**Keystone Backend container**
Connected to my postgress container
Has a docker volume mounted where file uploads are placed

**Postgress Database container**
Nothing special

**NextJS Frontend container**
Connects to KS Graphql via Apollo
Has a docker volume mounted where file uploads are available (mounted at `public/uploads/` to be precise

----

With that said, the problem I'm currently running in to is that NextJS doesn't allow images that are added to the public folder _after_ build time. So basically, with every Blogpost that's being made with files added I would need to restart the _NextJS Frontend container_. Obviously, that's super tedious.

I know this is not necessarily a Keystone issue, since the main thing causing this issue is my NextJS setup. Other than that, it would be great if someone has ideas, tips or advice on how to deal with images in a blog type situation.

**Comment by gwyneplaine** [comment, marked as answer]
@MatthijsMali we're currently reworking how images and files works in #7070. As part of that Keystone can server the images for you and return absolute urls to the stored image so your frontend can get them.

**Reply by MatthijsMali** [reply]
Thanks, I'll give it a bit more wait for now, and won't dive into the whole blobbing thing ;-)
